File: baitaptuan3/Julia/chuyendoisohe.jl
function decimal_to_binary(n)
    stack = Int[]
    println("Chuyen doi so thap phan: $n sang nhi phan.")
    
    if n == 0
        push!(stack, 0)
        println("Da them phan du '0' vao ngan xep.")
    end

    while n > 0
        remainder = n % 2
        push!(stack, remainder)
        println("Chia $n cho 2, phan du: $remainder → Da them $remainder vao ngan xep.")
        n = div(n, 2)
        println("Thuong moi: $n")
    end

    binary = ""
    println("Chuyen doi ngan xep thanh so nhi phan:")
    while !isempty(stack)
        push!(binary, string(pop!(stack)))
        println("Lay $(last(binary)) tu ngan xep va noi vao ket qua.")
    end

    return join(reverse(binary))
end

# Minh họa
number = 13
binary = decimal_to_binary(number)
println("So thap phan $number chuyen sang nhi phan la $binary")  # Output: 1101

File: baitaptuan3/C++/chuyendoisohe.cpp
#include <iostream>
#include <stack>
#include <vector>
#include <algorithm>
using namespace std;

string decimal_to_binary(int n) {
    stack<int> stack;
    cout << "Chuyen doi so thap phan: " << n << " sang nhi phan." << endl;

    if (n == 0) {
        stack.push(0);
        cout << "Da them phan du '0' vao ngan xep." << endl;
    }

    while (n > 0) {
        int remainder = n % 2;
        stack.push(remainder);
        cout << "Chia " << n << " cho 2, phan du: " << remainder << " → Da them " << remainder << " vao ngan xep." << endl;
        n = n / 2;
        cout << "Thuong moi: " << n << endl;
    }

    string binary = "";
    cout << "Chuyen doi ngan xep thanh so nhi phan:" << endl;
    while (!stack.empty()) {
        binary += to_string(stack.top());
        cout << "Lay " << stack.top() << " tu ngan xep va noi vao ket qua." << endl;
        stack.pop();
    }

    return binary;
}

int main() {
    int number = 13;
    string binary = decimal_to_binary(number);
    cout << "So thap phan " << number << " chuyen sang nhi phan la " << binary << endl;  // Output: 1101
    return 0;
}
